### ISSUE TITLE
Refactoring GFG repo 

### ISSUE THREAD

**Issue by kohchihao** [GiftForGood/website#426]
# Issue
As GFG repo continues to grow larger, the current folder structure doesn't seem be as clear and direct to a developer. 

As a result, differentiating from a `dummy` component from a `page` component is difficult. 

# Suggestion
`src/components`: 
- Reserved only for `dummy` components
- No api/business logic should be found within such components
- Components here should be generic and not tied to any particular page

`src/pages`:
- Reserved only for `page` components
- Business/Apis can be found here
- `src/pages/{pageName}/components` 
  - Components that belong to this `page` components should live within here
  - This can be a recursive folder structure. 
    - Eg: `src/pages/components/RegisterDonor/index.js` can have `src/pages/components/RegisterDonor/components/RegisterCard/index.js`  
- `src/pages/{pageName}/utils`
  - utils that belong to this page should live here
- `src/pages/{pageName}/constants`
  - constants that belong to this page should live here 
`src/utils`
 - Reserved only for generic utils
`src/constants` 
 - Reserved only for generic constants 
 
 # Benefits
 - Separation of concern from each page. As such, we can introduce new UI libraries for individual pages without worrying that it might affect other pages. 
 - Easier to scale in terms of add new pages

**Comment by kohchihao** [comment]
Please share your concerns and suggestions if any

**Comment by jamessspanggg** [comment]
I agree that the above is a more maintainable structure, but refactoring this might take quite an effort so I suggest that we do this incrementally, perhaps on a page by page basis

**Comment by kohchihao** [comment]
@jamessspanggg Agreed. This should be done incrementally. After this is approved by everyone, I will make a todo list within this issue to track what needs to changed.

**Comment by kohchihao** [comment]
@jinyingtan  

Regarding 
1. Yes. In some sense. Stuff that can be within here could be our `Cards`, `Buttons`, `Footer`, etc... 
2. Could be our current `utils/algolia/filteringRules` --> if we can split them into respective pages, it much easier for the developer to find.
3. Yes. Colors will be within generic constants

**Comment by kohchihao** [comment]
After speaking to everyone. We have decided to go ahead with this major refactoring. I will put a Todo list inside this issue to keep track of the changes that we need to make over the subsequent weeks.

**Comment by kohchihao** [comment]
Let's refactor pages first and leave the individual components alone. 

By doing this refactor, there might be duplicated code. It is fine because it helps with the readability of the codebase overall. But we shouldn't be afraid of having some duplicated code.
